Framework/Algorithms/src/CalculateIqt.cpp

```diff
@@ -0,0 +1,309 @@
+  return {"TransformToIqt"};
+}
+
+const std::string CalculateIqt::category() const {
+  return "Inelastic\\Indirect";
+}
+
+const std::string CalculateIqt::summary() const {
+  return "Calculates I(Q,t) from S(Q,w) and computes the errors using a "
+         "monte-carlo routine.";
+}
+
+void CalculateIqt::init() {
+
+  declareProperty(
+      make_unique<WorkspaceProperty<>>("InputWorkspace", "", Direction::Input),
+      "The name of the sample workspace.");
+  declareProperty(make_unique<WorkspaceProperty<>>("ResolutionWorkspace", "",
+                                                   Direction::Input),
+                  "The name of the resolution workspace.");
+
+  declareProperty("EnergyMin", -0.5,
+                  "Minimum energy for fit. Default = -.0.5.");
+  declareProperty("EnergyMax", 0.5, "Maximum energy for fit. Default = 0.5.");
+  declareProperty("EnergyWidth", 0.1, "Width of energy bins for fit.");
+
+  auto positiveInt = boost::make_shared<Kernel::BoundedValidator<int>>();
+  positiveInt->setLower(1);
+
+  declareProperty("NumberOfIterations", DEFAULT_ITERATIONS, positiveInt,
+                  "Number of randomised simulations within error to run.");
+  declareProperty(
+      "SeedValue", DEFAULT_SEED, positiveInt,
+      "Seed the random number generator for monte-carlo error calculation.");
+
+  declareProperty(make_unique<WorkspaceProperty<>>("OutputWorkspace", "",
+                                                   Direction::Output),
+                  "The name to use for the output workspace.");
+}
+
+void CalculateIqt::exec() {
+  const auto rebinParams = rebinParamsAsString();
+  const MatrixWorkspace_sptr sampleWorkspace = getProperty("InputWorkspace");
+  MatrixWorkspace_sptr resolution = getProperty("ResolutionWorkspace");
+  resolution = normalizedFourierTransform(resolution, rebinParams);
+
+  // create function which only needs sampleWorkspace as input so it can be
+  // easily repeated in simulations
+  auto calculateIqtFunction =
+      [resolution, &rebinParams](MatrixWorkspace_sptr sample) {
+        return calculateIqt(sample, resolution, rebinParams);
+      };
+
+  auto outputWorkspace = monteCarloErrorCalculation(sampleWorkspace, resolution,
+                                                    calculateIqtFunction);
+
+  outputWorkspace = cleanOutput(outputWorkspace);
+  setProperty("OutputWorkspace", outputWorkspace);
+}
+
+std::string CalculateIqt::rebinParamsAsString() {
+  const double e_min = getProperty("EnergyMin");
+  const double e_max = getProperty("EnergyMax");
+  const double e_width = getProperty("EnergyWidth");
+  return createRebinString(e_min, e_max, e_width);
+}
+
+MatrixWorkspace_sptr CalculateIqt::monteCarloErrorCalculation(
+    MatrixWorkspace_sptr sample, MatrixWorkspace_sptr resolution,
+    const std::function<MatrixWorkspace_sptr(MatrixWorkspace_sptr)> &
+        calculateIqtFunction) {
+  auto outputWorkspace = calculateIqtFunction(sample);
```

Comment:
As mentioned previously, it would be good to remove the use of std::function here.